src/main/scala/io/github/openDFDL/IBM_DFDL.scala

```diff
@@ -0,0 +1,377 @@
+        // System.err.println("Using Daffodil Entity Resolver")
+        grammarFactory.buildGrammarFromSchema(schemaUri, er)
+      } catch {
+        case e: DFDLException => {
+          toss(e)
+        }
+      }
+    if (grammar == null) {
+      Left(getDiagnostics)
+    } else {
+      Right((getDiagnostics, new IBMTDMLDFDLProcessor(diagnostics, grammar, bindings, optRootName, rootNamespace)))
+    }
+  }
+
+  private lazy val compileErrorHandler = new DFDLErrorHandler(IBMDFDLMode.Compile)
+}
+
+trait DiagnosticsMixin {
+
+  def isError: Boolean = diagnostics.exists { _.isError }
+
+  def getDiagnostics: Seq[Diagnostic] = diagnostics
+
+  protected var diagnostics: Seq[IBMTDMLDiagnostic] = Seq()
+
+  class DFDLErrorHandler(mode: IBMDFDLMode.Type) extends IDFDLProcessorErrorHandler with ErrorHandler {
+
+    override def processingError(diagnostic: IDFDLDiagnostic): Unit = {
+      diagnostics = new IBMTDMLDiagnostic(diagnostic, mode) +: diagnostics
+    }
+
+    override def schemaDefinitionError(diagnostic: IDFDLDiagnostic): Unit = {
+      diagnostics = new IBMTDMLDiagnostic(diagnostic, mode) +: diagnostics
+    }
+
+    override def validationError(diagnostic: IDFDLDiagnostic): Unit = {
+      diagnostics = new IBMTDMLDiagnostic(diagnostic, mode) +: diagnostics
+    }
+
+    override def warning(diagnostic: IDFDLDiagnostic): Unit = {
+      diagnostics = new IBMTDMLDiagnostic(diagnostic, mode) +: diagnostics
+    }
+
+    /*
+     * These methods make this a sax ErrorHandler
+     */
+
+    override def error(spe: SAXParseException): Unit = {
+      processingError(new IDFDLDiagFromThrowable(spe))
+    }
+
+    override def warning(spe: SAXParseException): Unit = {
+      warning(new IDFDLDiagFromThrowable(spe))
+    }
+
+    override def fatalError(spe: SAXParseException): Unit = {
+      processingError(new IDFDLDiagFromThrowable(spe))
+    }
+  }
+}
+
+class IBMTDMLDFDLProcessor(
+  compilerDiags: Seq[IBMTDMLDiagnostic],
+  grammar: IDFDLGrammar,
+  bindings: Seq[Binding],
+  optRootName: Option[String],
+  rootNamespace: String)
+  extends TDMLDFDLProcessor
+  with DiagnosticsMixin {
+
+  private def processorFactory = new DFDLProcessorFactory
+
+  val DFDL_NAMESPACE = "http://www.ogf.org/dfdl/dfdl-1.0/"
+
+  private var shouldValidate: Boolean = false
+
+  override def setValidationMode(validationMode: ValidationMode.Type): Unit = {
+    shouldValidate = validationMode match {
+      case ValidationMode.Full => true
+      case ValidationMode.Limited => true
+      case ValidationMode.Off => false
+    }
+  }
+
+  override def parse(is: java.io.InputStream, lengthLimitInBits: Long): TDMLParseResult = {
+
+    val parser: IDFDLParser = processorFactory.createParser
+    parser.setGrammar(grammar)
+    parser.setRootElement(
+      optRootName.getOrElse(Assert.usageError("Must call setDistinguishedRootNode before parse.")),
+      rootNamespace)
+
+    parser.setVariable("encoding", DFDL_NAMESPACE, "ISO-8859-1")
```

Comment:
These are wrong I think.  These shouldn't be hard-coded here. Should come from config files or embedded configs in the TDML.